### ISSUE TITLE
Quickchoice - allow visual cards to work with checkbox fields

### ISSUE THREAD

**Issue by lemerv** [alexed1/LightningFlowComponents#1238]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
The visual cards look great in flows. It would be awesome if these worked for checkbox fields too. For example:

- I have three checkbox field on an object
- I want to display each checkbox on a flow screen for a user to check/uncheck as they like
- This would be one visual card per checkbox
- The visual card could have a default value specified (ie true or false)
- A user could click the visual card and it would toggle between ticked/unticked (true/false)
- Otherwise, all the same functionality as is currently available with visual cards (ie: a label, description text, optional icons, etc)

**Describe alternatives you've considered**
I've tried to use the current component to do this and I can get as far as displaying the visual card. But, it doesn't toggle on clicks. If you click it so it is "ticked" then the only way to untick is to select a different visual card, but I can't because there is only one choice in the collection

**Additional context**
None

**Comment by ericrsmith35** [comment]
How about trying a flow screen with 2 visual cards for each checkbox field?  One true and the other false.  Then use conditional visibility to display one or the other.

**Comment by lemerv** [comment]
Hi Eric, not sure how that would work? Let's say the checkbox is currently true, so I display the "ticked" visual card (and hide the "unticked" one). How would I then change the checkbox to false? The visual card, once "ticked" cannot be unticked. The only way to untick it is to select a different visual card that is part of the same collection of choices (as far as I can see).

**Comment by ericrsmith35** [comment]
Each card would display as "unticked" with the opposite logic of the other
one.  Example:

CARD A - Click to Select CARD B
CARD B - Click to Select CARD A

**Comment by lemerv** [comment]
Still not following sorry. Do you mean neither card would ever display in a "ticked" state (ie: tick mark in top right and blue outline)? If so how would the user know if their selection is true or false?

**Comment by ericrsmith35** [comment]
You may be better suited with the standard Toggle component here.

**Comment by lemerv** [comment]
Yep that would work. But I'm interested in the visual cards for the aesthetics and layout. Eg: a label heading, with description text underneath, optional icons.

**Comment by ericrsmith35** [comment]
Another option would be to always display 2 cards, of which only one is selected at a time rather than a single card with a "toggle"

**Comment by lemerv** [comment]
Sorry @ericrsmith35 been on holiday. Can't get to work using the various work arounds suggested. 

I think there are some good use-cases to make it worthwhile allowing this to work specifically for checkboxes, but understand it might be a lot of work. If you're happy to try, I will help! However, no worries if not.

Thanks for all the replies :)

**Comment by ericrsmith35** [comment]
I've mentioned this use case to another developer that is currently working
on a few enhancements and he will try and add this capability.

**Comment by lemerv** [comment]
Great, thanks :)
